src/main/java/com/thebois/models/world/World.java

```diff
@@ -31,7 +30,7 @@ public World(int worldSize) {
         }
 
         // Structures
-        structuresMatrix = new Structure[worldSize][worldSize];
+        structuresMatrix = new House[worldSize][worldSize];
```

Comment:
This is not a good way of instantiating the matrix. I can see why you used `House` as both `IStructure` and `AbstractStructure` can't be instantiated, but I think you can still create an `IStructure[worldSize][worldSize]`. Or am I wrong? 

If it is not possible, then we need to rethink this, because the matrix needs to be able to hold `Structure`s other than `House`s.

Reply:
IStructure probably works, I was a bit tired when refactoring, will take a look at this tomorrow.